Feature/action combat movement

Jinyaden opened the pull request:
THIS IS SO NASTY

Jinyaden committed:
Action Combat shit

Added in a few of the scripts I was working on

Jinyaden committed:
Update#1

Added in player rotate functionality, require camera to rotate with player

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCamera.cs:
```diff
@@ -0,0 +1,39 @@
+
+public class trdprsnCamera : MonoBehaviour
+{
+    [SerializeField] Transform lookAt;
```
Doesn't match the private field naming convention of the project. Also missing an explicit access modifier. Please rectify both.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCamera.cs:
```diff
@@ -0,0 +1,39 @@
+public class trdprsnCamera : MonoBehaviour
+{
+    [SerializeField] Transform lookAt;
+    [SerializeField] Transform camTransform; // DELETE THIS
```
Doesn't match the private field naming convention of the project. Also missing an explicit access modifier. Please rectify both.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCamera.cs:
```diff
@@ -0,0 +1,39 @@
+
+    private float distance = 15.0f;
+    private float currentX = 0.0f;
+    private float currentY = 0.0f;
```
None of the above follow the private field naming convention of the project.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCamera.cs:
```diff
@@ -0,0 +1,39 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class trdprsnCamera : MonoBehaviour
```
This class name breaks the naming convention for classes. Please rectify. Also this name is double dutch, the hell is `trdprsn`?

Banananaman91 replied:
Turd Person, or Irish for "Third Person"

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCharacter.cs:
```diff
@@ -0,0 +1,83 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class trdprsnCharacter : MonoBehaviour
```
These class names, plsfix.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCharacter.cs:
```diff
@@ -0,0 +1,83 @@
+    [SerializeField] float rotatespeed;
+    //private float rotX;
+    //private float rotY;
+    [SerializeField] float rotZ;
```
all of the above private fields do not meet the convention of the codebase. Also missing access modifiers.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCharacter.cs:
```diff
@@ -0,0 +1,83 @@
+    //private float rotX;
+    //private float rotY;
+    [SerializeField] float rotZ;
+    private Camera cam;
```
all of the above private fields do not meet the convention of the codebase.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCharacter.cs:
```diff
@@ -0,0 +1,83 @@
+    void Update()
+    {
+        Playerturn(); // turn to face the player
+        Playerrotate(); // rotate the player itself
```
Please fix these method names, they do not match the convention of the codebase.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCharacter.cs:
```diff
@@ -0,0 +1,83 @@
+        Playermovement();
+    }
+
+    void Playermovement()
```
Please fix this method name, it does not match the convention of the codebase.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCharacter.cs:
```diff
@@ -0,0 +1,83 @@
+        {
+            transform.position += transform.TransformDirection(Vector3.left) * Time.deltaTime * movespeed;
+        }
+        else if (Input.GetKey("d") && !Input.GetKey("a")) // if only d is pressed
```
All of the above `GetKey` calls rely on strings and not the `KeyCode` enum. Please rectify this.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCharacter.cs:
```diff
@@ -0,0 +1,83 @@
+            transform.Rotate(new Vector3(0, 0, rotZ) * Time.deltaTime * rotatespeed);
+        }
+        // if r is pressed rotate body right
+        if (Input.GetKey("e"))
```
All of the above GetKey calls rely on strings and not the KeyCode enum. Please rectify this.

RubyNova reviewed (changes requested):
I've not seen anything that's gonna brick your game on this first pass but a lot of the naming alienated me, so maybe with a second pass I'll see something.

Banananaman91 committed:
Code Review, changed naming

Renamed items to abide to coding convention

Jinyaden committed:
Update#2

My PP Hard

Jinyaden committed:
Merge branch 'feature/action-combat-movement' of https://github.com/Banananaman91/Hooked-Reel-Life into feature/action-combat-movement

Jinyaden committed:
Update#3

Added Rotation options, still in progress

Jinyaden committed:
Update #idgaf

2 Issues remain, Making the camera rotate with the player when pressing Q and E. second issue is the controls get inverted when mouse position hits 0,0,0 or lower

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCameraController.cs:
```diff
@@ -0,0 +1,46 @@
+
+    private Camera _cam;
+
+    private float _distance = 30.0f;    // DISTANCE TO STAY BEHIND PLAYER
```
This might be better as a serializable, configurable value instead of just a hard 30.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+    //private float rotX;
+    //private float rotY;
+    [SerializeField] private float _rotZ;
+    private Camera _cam;
```
Abbreviations are generally frowned upon. Consider using proper names (same goes for `_rotZ` and other such variables).

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+        {
+            transform.Rotate(new Vector3(0, 0, _rotZ) * Time.deltaTime * _rotateSpeed);
+        }
+        // if r is pressed rotate body right
```
This comment does not match the code. Either remove the comment or update it.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+
+
+
+        void PlayerTurn() // EXPERIMENTING ATM
```
Looks like tabs and spaces got mixed here. Plsfix.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+    private void FixedUpdate()
+    {
+
+        PlayerMovement();
```
User input should be handled in `Update` not `FixedUpdate`.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCameraController.cs:
```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ThirdPersonCameraController : MonoBehaviour
```
This class is currently in the `global::` namespace. Please provide a proper namespace.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ThirdPersonCharacterController : MonoBehaviour
```
This class is currently in the global:: namespace. Please provide a proper namespace.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+    void Update()
+    {
+
+        PlayerTurn(); // turn to face the player
```
These comments leave much to be desired. Please consider making the method names more concise instead.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+        //rotY += Input.GetAxis("Mouse X") * Time.deltaTime * rotatespeed;
+
+        //transform.rotation = Quaternion.Euler(rotX, rotY, 0);
+        //cam.transform.rotation = Quaternion.Euler(rotX, rotY, 0);
```
Is all this dead code needed?

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCameraController.cs:
```diff
@@ -0,0 +1,46 @@
+        // _camTransform.RotateAround(Vector3.zero, Vector3.up, 20 * Time.deltaTime);       /* da fuq even is dis? */
+        // _camTransform.rotation = _lookAt.localRotation;      /* This works but obvs not what is needed */
+        // Vector3 _targetPosition = new Vector3(0, 0, _camTransform.localRotation.z);   _camTransform.LookAt(_targetPosition);
+        // _camTransform.RotateAround (_camTransform.position, _camTransform.up, _lookAt.rotation.z); /* idk what I'm doing anymore */
```
All this dead code, please remove.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+    void PlayerMovement()
+    {
+        //forward and back controls
+        if (Input.GetKey(KeyCode.LeftShift) && Input.GetKey(KeyCode.W)) // if w and shift are pressed
```
might I advise a refactor into a proper state machine?

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+    void PlayerRotate()
+    {
+        // if q is pressed rotate body left
+        if (Input.GetKey(KeyCode.Q))
```
This can be made a one-line if.

RubyNova commented on GameProject/Assets/OpenWorld Assets/Scripts/ThirdPersonCharacterController.cs:
```diff
@@ -0,0 +1,88 @@
+            transform.Rotate(new Vector3(0, 0, _rotZ) * Time.deltaTime * _rotateSpeed);
+        }
+        // if r is pressed rotate body right
+        if (Input.GetKey(KeyCode.E))
```
This can be made a one-line if.
